### ISSUE TITLE
Peers are saved with the proxy's IP

### ISSUE THREAD

**Issue by SomberNight** [kyuupichan/electrumx#301]
On #electrum IRC, skace reported that:

He has `FORCE_PROXY=1` on his elec.luggs.co server instance. He uses a local Tor proxy (at 127.0.0.1).

His output for `$ electrumx_rpc.py peers`

<details>
 <summary>peers</summary>

```
Host                           Status   TCP   SSL Server             Min  Max  Pruning   Last Good    Last Try Tries               Source IP Address
bauerjhejlv6di7s.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1              08m 13s     08m 23s     0                 peer
helicarrier.bauerj.eu          good   50001 50002 ElectrumX 1.2      0.9  1.1              08m 16s     08m 23s     0                 peer 127.0.0.1
bitcoin.freedomnode.com        good   50001 50002 ElectrumX 1.2      0.9  1.1              42m 55s     42m 58s     0                 peer 127.0.0.1
alviss.coinjoined.com          good   50001 50002 ElectrumX 1.2      0.9  1.1              42m 57s     42m 58s     0                 peer 127.0.0.1
j5jfrdthqt5g25xz.onion         good   50001 50002 ElectrumX 1.2      1.0  1.0              58m 25s     58m 35s     0                 peer
electrum.be                    good   50001 50002 ElectrumX 1.1      0.9  1.1          01h 33m 01s 01h 33m 03s     0                 peer 127.0.0.1
electrum3.hachre.de            good   50001 50002 ElectrumX 1.2      0.9  1.1          01h 45m 02s 01h 45m 04s     0                 peer 127.0.0.1
electrum.ezr.se                good   50001 50002 ElectrumX 1.1.1    0.9  1.1          02h 08m 42s 02h 08m 44s     0                 peer 127.0.0.1
node.xbt.eu                    good   50001 50002 ElectrumX 1.1      0.9  1.1          02h 33m 31s 02h 33m 44s     0                 peer 127.0.0.1
s7clinmo4cazmhul.onion         good   50001       ElectrumX 1.2      0.9  1.1          02h 48m 44s 02h 48m 53s     0                 peer
e-1.claudioboxx.com            good   50001 50002 ElectrumX 1.2      0.9  1.1          02h 48m 51s 02h 48m 53s     0                 peer 127.0.0.1
electrum-server.ninja          good   50001 50002 ElectrumX 1.2      0.9  1.1          03h 09m 30s 03h 10m 24s     0                 peer 127.0.0.1
o6nvntbo3qsn36dm.onion         good   50001 50002 ElectrumX 1.0.17   1.0  1.0          03h 15m 16s 03h 15m 24s     0                 peer
bitcoins.sk                    good   50001 50002 ElectrumX 1.2      0.9  1.1          03h 55m 49s 03h 55m 50s     0                 peer 127.0.0.1
165.227.22.180                 good   50001 50002 ElectrumX 1.1      0.9  1.1          04h 05m 50s 04h 06m 23s     0                 peer 127.0.0.1
ELECTRUMX.not.fyi              good   50001 50002 ElectrumX 1.2      0.9  1.1          04h 51m 23s 04h 51m 26s     0                 peer 127.0.0.1
node.ispol.sk                  good   50001 50002 ElectrumX 1.1.1    0.9  1.1          04h 51m 23s 04h 51m 26s     0                 peer 127.0.0.1
electrum.taborsky.cz           good         50002 ElectrumX 1.0.17   1.0  1.0          05h 26m 26s 05h 26m 30s     0                 peer 127.0.0.1
tor.fh-biergarten.de           good   50001 50002 ElectrumX 1.1      0.9  1.1          05h 26m 27s 05h 26m 30s     0                 peer 127.0.0.1
hsmiths5mjk6uijs.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          06h 16m 20s 06h 16m 30s     0                 peer
4cii7ryno5j3axe4.onion         good   50001       ElectrumX 1.1.2    0.9  1.1          06h 21m 30s 06h 21m 36s     0                 peer
electrum.vforge.net            good   50001 50002 ElectrumX 1.1.1    0.9  1.1          06h 21m 33s 06h 21m 36s     0                 peer 127.0.0.1
134.119.179.55                 good         50002 ElectrumX 1.0.17   1.0  1.0          06h 31m 36s 06h 31m 38s     0                 peer 127.0.0.1
electrum.draylin.de            good   50001 50002 ElectrumX 1.1      0.9  1.1          06h 41m 35s 06h 41m 38s     0                 peer 127.0.0.1
btcelx7bagcvnptz.onion         good   50001       ElectrumX 1.1.1    0.9  1.1          06h 46m 33s 06h 46m 38s     0                 peer
4yi77lkjgy4bwtj3.onion         good   50001 50002 ElectrumX 1.0.17   1.0  1.0          06h 51m 26s 06h 51m 38s     0                 peer
ozahtqwp25chjdjd.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          06h 51m 30s 06h 51m 38s     0                 peer
arihancckjge66iv.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          06h 51m 30s 06h 51m 38s     0                 peer
electrum.dk                    good   50001 50002 ElectrumX 1.0.17   1.0  1.0          06h 51m 35s 06h 51m 38s     0                 peer 127.0.0.1
s4.noip.pl                     good         50104 ElectrumX 1.2      0.9  1.1          06h 55m 36s 06h 56m 38s     0                 peer 127.0.0.1
s2.noip.pl                     good         50102 ElectrumX 1.2      0.9  1.1          06h 55m 36s 06h 56m 38s     0                 peer 127.0.0.1
s1.noip.pl                     good         50101 ElectrumX 1.2      0.9  1.1          06h 55m 36s 06h 56m 38s     0                 peer 127.0.0.1
s3.noip.pl                     good         50103 ElectrumX 1.2      0.9  1.1          06h 55m 36s 06h 56m 38s     0                 peer 127.0.0.1
s5.noip.pl                     good         50105 ElectrumX 1.2      0.9  1.1          06h 55m 37s 06h 56m 38s     0                 peer 127.0.0.1
hsmiths4fyqlw5xw.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          06h 56m 14s 06h 56m 38s     0                 peer
shsmithsj5cjphxs.onion         good   51001 51002 ElectrumX 1.2      0.9  1.1          06h 56m 27s 06h 56m 38s     0                 peer
bauerjda5hnedjam.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          06h 56m 30s 06h 56m 38s     0                 peer
he36kyperp3kbuxu.onion         good   50001 50002 ElectrumX 1.1      0.9  1.1          06h 56m 30s 06h 56m 38s     0                 peer
3tm3fjg3ds5fcibw.onion         good   50001       ElectrumX 1.2      0.9  1.1          06h 56m 32s 06h 56m 38s     0                 peer
mooo.not.fyi                   good   50011 50012 ElectrumX 1.1.1    0.9  1.1          06h 56m 35s 06h 56m 38s     0                 peer 127.0.0.1
raspi.hsmiths.com              good   51001 51002 ElectrumX 1.2      0.9  1.1          06h 56m 35s 06h 56m 38s     0                 peer 127.0.0.1
currentlane.lovebitco.in       good   50001       ElectrumX 1.2      0.9  1.1          06h 56m 36s 06h 56m 38s     0                 peer 127.0.0.1
btc1.bitwage.com               good   50001       1.0                1.0  1.0      100 06h 56m 36s 06h 56m 38s     0                 peer 127.0.0.1
bitcoin3nqy3db7c.onion         good   50001 50002 ElectrumX 1.1      0.9  1.1          06h 59m 43s 07h 01m 42s     0                 peer
elec.luggs.co                  good           443 ElectrumX 1.2      0.9  1.1          07h 01m 20s 07h 01m 42s     0                 peer 127.0.0.1
gibizbemyzskqsmb.onion         good   50001 50002 1.1/j/absurd      0.10  1.1          07h 01m 29s 07h 01m 42s     0                 peer
luggscoqbymhvnkp.onion         good      80       ElectrumX 1.2      0.9  1.1          07h 01m 29s 07h 01m 42s     0                 peer
ndndword5lpb7eex.onion         good   50001       ElectrumX 1.2      0.9  1.1          07h 01m 30s 07h 01m 42s     0                 peer
electrumxhqdsmlu.onion         good   50001       ElectrumX 1.2      0.9  1.1          07h 01m 31s 07h 01m 42s     0                 peer
bauerjsb3k4orv2h.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 33s 07h 01m 42s     0                 peer
electrum.mindspot.org          good   50001 50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 35s 07h 01m 42s     0                 peer 127.0.0.1
icarus.tetradrachm.net         good   50001 50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 35s 07h 01m 42s     0                 peer 127.0.0.1
electrum.backplanedns.org      good   50001 50002 ElectrumX 1.1      0.9  1.1          07h 01m 36s 07h 01m 42s     0                 peer 127.0.0.1
electrum2.everynothing.net     good   50001 50002 ElectrumX 1.1.1    0.9  1.1          07h 01m 36s 07h 01m 42s     0                 peer 127.0.0.1
electrumx.nmdps.net            good   50001 50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 36s 07h 01m 42s     0                 peer 127.0.0.1
electrum2.villocq.com          good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 36s 07h 01m 42s     0                 peer 127.0.0.1
node.arihanc.com               good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 36s 07h 01m 42s     0                 peer 127.0.0.1
elx01.knas.systems             good   50001 50002 ElectrumX 1.1.1    0.9  1.1          07h 01m 36s 07h 01m 42s     0                 peer 127.0.0.1
daedalus.bauerj.eu             good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 37s 07h 01m 42s     0                 peer 127.0.0.1
electrum.no-ip.org             good   50001 50002 1.0                1.0  1.0    10000 07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.akinbo.org            good   50001 50002 ElectrumX 1.1      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.infinitum-nihil.com   good         50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.hsmiths.com           good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
orannis.com                    good   50001 50002 ElectrumX 1.1      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrumx.bot.nu               good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
btc.smsys.me                   good           995 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
bordens.us                     good   50001 50002 ElectrumX 1.1      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.coinucopia.io         good   50001 50002 ElectrumX 1.1      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
ele.nummi.it                   good         50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
46.166.165.18                  good         50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.antumbra.se           good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrumx.soon.it              good   50001 50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
185.64.116.15                  good   50001 50002 ElectrumX 1.1.1    0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.qtornado.com          good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum0.snel.it              good   50001 50002 ElectrumX 1.0.17   1.0  1.0          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
ndnd.selfhost.eu               good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
kirsche.emzy.de                good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
VPS.hsmiths.com                good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 38s 07h 01m 42s     0                 peer 127.0.0.1
electrum.villocq.com           good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 39s 07h 01m 42s     0                 peer 127.0.0.1
songbird.bauerj.eu             good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 39s 07h 01m 42s     0                 peer 127.0.0.1
electrum.vom-stausee.de        good   50001 50002 ElectrumX 1.1.2    0.9  1.1          07h 01m 39s 07h 01m 42s     0                 peer 127.0.0.1
electrumx.adminsehow.com       good   50001 50002 ElectrumX 1.2      0.9  1.1          07h 01m 39s 07h 01m 42s     0                 peer 127.0.0.1
h.1209k.com                    good   50011 50012 1.1/j/absurd      0.10  1.1          08h 46m 42s 08h 46m 46s     0                 peer 127.0.0.1
us.electrum.be                 good   50001 50002 ElectrumX 1.1      0.9  1.1          09h 01m 46s 09h 01m 50s     0                 peer 127.0.0.1
176.9.155.246                  good         50002 ElectrumX 1.2      0.9  1.1          09h 01m 49s 09h 01m 50s     0                 peer 127.0.0.1
electrum.festivaldelhumor.org  good   50001 50002 ElectrumX 1.2      0.9  1.1          09h 26m 50s 09h 27m 01s     0                 peer 127.0.0.1
electrumx.donsomhong.net       good   50001 50002 ElectrumX 1.2      0.9  1.1          10h 12m 01s 10h 12m 02s     0                 peer 127.0.0.1
k5fjhs6f374wi4zh.onion         good         50105 ElectrumX 1.2      0.9  1.1          10h 46m 56s 10h 47m 03s     0                 peer
e.keff.org                     good   50001 50002 ElectrumX 1.2      0.9  1.1          10h 47m 02s 10h 47m 03s     0                 peer 127.0.0.1
us01.hamster.science           good   50001 50002 ElectrumX 1.2      0.9  1.1          11h 17m 03s 11h 17m 05s     0                 peer 127.0.0.1
gttvzmfh23pxqiw4.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          11h 57m 05s 11h 57m 11s     0                 peer
e-3.claudioboxx.com            good   50001 50002 ElectrumX 1.2      0.9  1.1          12h 02m 11s 12h 02m 13s     0                 peer 127.0.0.1
bitcoin.cluelessperson.com     good   50001 50002 ElectrumX 1.2      0.9  1.1          12h 16m 55s 12h 17m 13s     0                 peer 127.0.0.1
E-X.not.fyi                    good   50001 50002 ElectrumX 1.2      0.9  1.1          14h 08m 30s 14h 08m 31s     0                 peer 127.0.0.1
ph3d32vdfojn45hn.onion         good         50103 ElectrumX 1.2      0.9  1.1          14h 18m 21s 14h 18m 31s     0                 peer
krvydnxj567slsag.onion         good         50102 ElectrumX 1.2      0.9  1.1          14h 18m 23s 14h 18m 31s     0                 peer
xmzotbhzstu5rhqr.onion         good         50104 ElectrumX 1.2      0.9  1.1          14h 18m 23s 14h 18m 31s     0                 peer
4zwgpbfa6hnpse4m.onion         good         50101 ElectrumX 1.2      0.9  1.1          14h 18m 25s 14h 18m 31s     0                 peer
erbium1.sytes.net              good   50001 50002 ElectrumX 1.2      0.9  1.1          14h 18m 28s 14h 18m 31s     0                 peer 127.0.0.1
electrum.petrkr.net            good   50001 50002 ElectrumX 1.0.17   1.0  1.0          14h 28m 31s 14h 28m 54s     0                 peer 127.0.0.1
37.187.167.216                 good         50002 ElectrumX 1.2      0.9  1.1          15h 03m 52s 15h 03m 54s     0                 peer 127.0.0.1
electrum.chainhost.io          good         50002 ElectrumX 1.0.17   1.0  1.0          15h 48m 54s 15h 48m 55s     0                 peer 127.0.0.1
electrumx.kekku.li             good         50002 ElectrumX 1.2      0.9  1.1          16h 13m 53s 16h 13m 55s     0                 peer 127.0.0.1
electrum-uk.klaybox.xyz        good   50001 50002 ElectrumX 1.2      0.9  1.1          16h 43m 53s 16h 43m 55s     0                 peer 127.0.0.1
ultra-ecoelectrum.my-gateway.d good   50001 50002 ElectrumX 1.1      1.0  1.0      100 16h 48m 43s 16h 48m 55s     0                 peer 127.0.0.1
wurmyopgkskigv4l.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          16h 48m 46s 16h 48m 55s     0                 peer
cryptohead.de                  good         50002 ElectrumX 1.2      0.9  1.1          16h 53m 49s 16h 53m 55s     0                 peer 127.0.0.1
ex-btc.server-on.net           good   50001 50002 ElectrumX 1.2      0.9  1.1          16h 53m 53s 16h 53m 55s     0                 peer 127.0.0.1
lith.strangled.net             good         50002 ElectrumX 1.2      0.9  1.1          17h 08m 53s 17h 08m 55s     0                 peer 127.0.0.1
agapc7dhcqfjqpjb.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          17h 43m 55s 17h 44m 01s     0                 peer
lh.site.rambo.pro              good   50001 50002 ElectrumX 1.2      0.9  1.1          17h 54m 00s 17h 54m 01s     0                 peer 127.0.0.1
electrum.glowhost.com          good   50001 50002 ElectrumX 1.1.1    1.1  1.1          19h 17m 07s 19h 18m 08s     0                 peer 127.0.0.1
electrumx.westeurope.cloudapp. good   50001 50002 ElectrumX 1.1      0.9  1.1          19h 33m 09s 19h 33m 09s     0                 peer 127.0.0.1
gyn2vguc35viks2b.onion         good   50001       ElectrumX 1.2      0.9  1.1          20h 08m 02s 20h 08m 09s     0                 peer
qtornadoklbgdyww.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          20h 08m 04s 20h 08m 09s     0                 peer
qmebr.spdns.org                good   50001 50002 ElectrumX 1.2      0.9  1.1          20h 08m 07s 20h 08m 09s     0                 peer 127.0.0.1
rbx.curalle.ovh                good         50002 ElectrumX 1.2      0.9  1.1          20h 18m 09s 20h 18m 12s     0                 peer 127.0.0.1
enode.duckdns.org              good   50001 50002 ElectrumX 1.1.1    0.9  1.1          20h 18m 10s 20h 18m 12s     0                 peer 127.0.0.1
electrum-btccore.nithanim.me   good   50001 50002 ElectrumX 1.0.17   1.0  1.0          21h 32m 41s 21h 32m 53s     0                 peer 127.0.0.1
2y27fkedpau73lbw.onion         good   50001 50002 ElectrumX 1.2      0.9  1.1          22h 42m 53s 22h 43m 41s     0                 peer
e-2.claudioboxx.com            good   50001 50002 ElectrumX 1.2      0.9  1.1          22h 43m 37s 22h 43m 41s     0                 peer 127.0.0.1
noserver4u.de                  good   50001 50002 ElectrumX 1.2      0.9  1.1          23h 43m 30s 23h 43m 31s     0                 peer 127.0.0.1
104.250.141.242                stale        50002 ElectrumX 1.2      0.9  1.1           1d 03h 20m     53m 35s     5                 peer 127.0.0.1
65732skw6wenqxap.onion         stale  50001 50002 ElectrumX 1.1.1    0.9  1.1           1d 04h 45m 01h 56m 44s     5                 peer
electrum.cutie.ga              stale  50001 50002 ElectrumX 1.2      0.9  1.1           1d 06h 55m 01h 48m 47s     6                 peer 127.0.0.1
electrum-btc.ddns.net          stale  50003 50004 ElectrumX 1.2      0.9  1.1           1d 19h 22m 08h 36m 29s     7                 peer 127.0.0.1
r63qdbkuoqbrr3zf.onion         stale        50105 ElectrumX 1.2      0.9  1.1           2d 06h 34m 09h 27m 01s     8                 peer
z67bk32zw5bz67ot.onion         stale        50102 ElectrumX 1.2      0.9  1.1           2d 06h 34m 09h 27m 01s     8                 peer
ah7v6oqjvw5pbha7.onion         stale        50104 ElectrumX 1.2      0.9  1.1           2d 06h 34m 09h 27m 01s     8                 peer
zy7eptqboryxj54i.onion         stale        50101 ElectrumX 1.2      0.9  1.1           2d 06h 39m 09h 32m 01s     8                 peer
ncjnym6ndwzf3rtc.onion         stale        50103 ElectrumX 1.2      0.9  1.1           2d 06h 39m 09h 32m 01s     8                 peer
electrum.anduck.net            stale  50001 50002 1.0                1.0  1.0      100  2d 08h 09m 10h 36m 53s     8                 peer 127.0.0.1
electrum.achow101.com          stale  50001 50002 ElectrumX 1.1.1    0.9  1.1           3d 16h 23m 21h 37m 49s     9                 peer 127.0.0.1
electrum.neocrypto.io          stale  50001 50002 ElectrumX 1.0.17   1.0  1.0           3d 16h 59m 22h 07m 12s     9                 peer 127.0.0.1
ruuxwv74pjxms3ws.onion         stale  50001 10042 ElectrumX 1.1.1    0.9  1.1           4d 06h 29m  1d 11h 46m     9                 peer
electrum.poorcoding.com        stale  50001 50002 ElectrumX 1.1.1    0.9  1.1           4d 06h 34m  1d 11h 51m     9                 peer 127.0.0.1
tomsplayground.de              stale  50001 50002 ElectrumX 1.2      0.9  1.1           6d 01h 29m  1d 12h 01m    10                 peer 127.0.0.1
electrumx.isimiz.web.tr        never  50601 50602 ElectrumX 1.1      0.9  1.1                Never     16m 08s     2         88.250.32.56
```
</details>

Note the last column which is full of localhost IPs.

To Electrum clients, when queried with `server.peers.subscribe`, an example reply is:
```
[elec.luggs.co] <-- {'result': [['luggscoqbymhvnkp.onion', 'luggscoqbymhvnkp.onion', ['v1.1', 't80']], ['127.0.0.1', 's2.noip.pl', ['v1.1', 's50102']], ['bauerjhejlv6di7s.onion', 'bauerjhejlv6di7s.onion', ['v1.1', 's50002', 't50001']], ['4zwgpbfa6hnpse4m.onion', '4zwgpbfa6hnpse4m.onion', ['v1.1', 's50101']], ['ozahtqwp25chjdjd.onion', 'ozahtqwp25chjdjd.onion', ['v1.1', 's50002', 't50001']], ['gttvzmfh23pxqiw4.onion', 'gttvzmfh23pxqiw4.onion', ['v1.1', 's50002', 't50001']], ['gyn2vguc35viks2b.onion', 'gyn2vguc35viks2b.onion', ['v1.1', 't50001']], ['bauerjsb3k4orv2h.onion', 'bauerjsb3k4orv2h.onion', ['v1.1', 's50002', 't50001']], ['j5jfrdthqt5g25xz.onion', 'j5jfrdthqt5g25xz.onion', ['v1.0', 's50002', 't50001']], ['127.0.0.1', 'electrum.antumbra.se', ['v1.1', 's50002', 't50001']], ['127.0.0.1', 'elec.luggs.co', ['v1.1', 's443']], ['qtornadoklbgdyww.onion', 'qtornadoklbgdyww.onion', ['v1.1', 's50002', 't50001']], ['o6nvntbo3qsn36dm.onion', 'o6nvntbo3qsn36dm.onion', ['v1.0', 's50002', 't50001']]], 'jsonrpc': '2.0', 'id': 3}
```

I'm guessing this reply is short as it's filtered to unique IPv4 /16 subnets, as in
https://github.com/kyuupichan/electrumx/blob/305fd897406e6f6609a033a7d633db4729b1b8e6/docs/PEER_DISCOVERY.rst#response-to-serverpeerssubscribe-rpc-call

> In ElectrumX, "recently" is taken to be the last 24 hours. Only one peer from each IPv4/16 netmask is returned, and the number of onion peers is limited.

Clearly, all "127.0.0.1" peers share a /16. :)

To sum up, it seems e-x saves all server peers with the Tor proxy's IP.

**Comment by luggs-co** [comment]
has this been looked at yet?

**Comment by kyuupichan** [comment]
I'm working on a rewrite of ElectrumX's RPC layer; see https://github.com/kyuupichan/aiorpcX.

When I do that I'll probably rework the proxy handling too.

**Comment by luggs-co** [comment]
That's great to see!

**Comment by kyuupichan** [comment]
Looking at the SOCKS5 spec there is no way for the proxy client to determine what the remote IP address is.  What do you suggest is the best thing to do here?

**Comment by luggs-co** [comment]
Maybe resolve the fqdn?

**Comment by kyuupichan** [comment]
@SomberNight - I have re-done the proxying.  Is this fixed now?

**Comment by kyuupichan** [comment]
Thanks.  To be sure I don't know as I don't run that setup, but if it's still broken it should now be easy to fix.  But I think I may have fixed it already.
